File: test_shopee_all/2.py
import pandas as pd

customer = pd.read_csv('test_shopee_all/Customer.csv')
customer = customer.groupby('City')['Customer ID'].unique().explode().reset_index(name = 'ID')

order = pd.read_csv('test_shopee_all/Orders.csv')

print(customer.head(5))
print(order.head(5))

hasil = pd.merge(customer, order, left_on= 'ID', right_on='Customer ID', how = 'inner')
hasil = hasil.groupby('City')['Order ID'].count().reset_index(name='jumlah').sort_values('jumlah', ascending=False).reset_index(drop=True)
print(hasil)

File: test_shopee_all/2.r
library(dplyr)

customer <- read.csv('test_shopee_all/Customer.csv')
customer <- customer %>% group_by(City) %>% summarise(ID = unique(Customer.ID))

order <- read.csv('test_shopee_all/Orders.csv')

print(head(customer))
print(head(order))
cat('\n\n\n')

# hasil <- merge(customer, order, by.x = 'ID', by.y = 'Customer.ID')
hasil <- inner_join(customer, order, by = c('ID' = 'Customer.ID'))
hasil <- hasil %>% group_by(City) %>% summarise(jumlah = n()) %>% arrange(desc(jumlah))
print(head(hasil))